app/controllers/predicates/AuthPredicate.scala

```diff
@@ -87,28 +87,48 @@ class AuthPredicate @Inject()(authService: EnrolmentsAuthService,
     } match {
       case Some(vrn) =>
         val user = User(vrn)
-        request.session.get(SessionKeys.insolventWithoutAccessKey) match {
-          case Some("true") => Future.successful(Forbidden(unauthorisedNonAgent()))
-          case Some("false") => block(user)
-          case _ => vatSubscriptionService.getCustomerDetails(user.vrn).flatMap {
-            case Right(details) if details.isInsolventWithoutAccess =>
-              Logger.debug("[AuthPredicate][checkVatEnrolment] - User is insolvent and not continuing to trade")
-              Future.successful(Forbidden(unauthorisedNonAgent())
-                .addingToSession(SessionKeys.insolventWithoutAccessKey -> "true"))
-            case Right(_) =>
-              Logger.debug("[AuthPredicate][checkVatEnrolment] - Authenticated as principle")
-              block(user).map(result => result.addingToSession(SessionKeys.insolventWithoutAccessKey -> "false"))
-            case _ =>
-              Logger.warn("[AuthPredicate][checkVatEnrolment] - Failure obtaining insolvency status from Customer Info API")
-              Future.successful(errorHandler.showInternalServerError)
-          }
+        (request.session.get(SessionKeys.insolventWithoutAccessKey), request.session.get(SessionKeys.futureInsolvencyBlock)) match {
```

Comment:
Because your code is only in the non-agent section of this predicate, an agent with a future insolvent client is able to use the service on behalf of them. The code you've added needs to move to a shared function/class so that it can be utilised for both principal users and agents.